src/frontend/src/components/SearchBar/SearchBar.jsx

```diff
@@ -166,11 +176,7 @@ function CustomizedInputBase(props) {
                 variant="outlined"
                 list="searchData"
               ></input>
-              <datalist id="searchData">
-                <option>Test 1</option>
-                <option>Test 2</option>
-                <option>Test 3</option>
-              </datalist>
+              <datalist id="searchData"></datalist>
```

Comment:
To do this properly, we're going to want to only apply the new `datalist` (and eventually fetch its data) if the user selects an `Author` filter.  The easiest way to achieve this is to use the `filter` value in order to decide whether or not to render this.  Another way to go would be to break this part of the search UI into its own components:

```jsx
<SearchInput filter={filter} />
```

And the `SearchInput.jsx` could do something like this:

```jsx
function AuthorSearchInput() {
  return (
    <>
      <input 
         //...rest of props here
          list="searchData"
      />
      <datalist id="searchData">...</datalist>
    </>
  );
}

function PostSearchInput() {
  return <input ... />
}

function SeachInput({ filter }) {
   return filter === 'author' ? <AuthorSearchInput /> : <PostSearchInput />;
}
```

Reply:
Ok, I think I see where your going here. What is the purpose of PostSearchInput() in this case?

Reply:
Where `AuthorSearchInput` includes a `<datalist>` and a `list="..."` attribute, `PostSearchInput` does not: it only returns the `<input ...>` where `...` is all of the attributes I didn't bother typing.

Reply:
Oh wait never mind, it would just be the output for a non autocomplete input.

Reply:
Exactly, and if you want, it could be what it is now vs being limited to what you had to do to get the `datalist` to work (i.e., they can use different components/elements to render the input).

Reply:
@humphd 

I have the jsx page setup and I can now switch between the two different author and post inputs. I'm now trying to figure out how make both these inputs to activate the SearchPage portion of the site.

Here's were I call my SearchInput.jsx providing the filter.

```jsx
<FormControl fullWidth={true}>
              <SearchInput filter={filter} />
</FormControl>
```

And then here's my AuthorSearchInput():

```jsx
function AuthorSearchInput(props) {
  const classes = useStyles();
  const { searchText, onChangeHandler } = props;

  const onTextChange = (event) => {
    onChangeHandler(event.target.value);
  };

  return (
    <>
      <input
        // ...rest of props here
        type="text"
        id="authorSearch"
        className={classes.input}
        placeholder="How to Get Started in Open Source"
        variant="outlined"
        list="searchData"
        value={searchText}
        onChange={(event) => onTextChange(event)}
        inputProps={{ 'aria-label': 'search telescope' }}
      />
      <datalist id="searchData">
        <option>Test 1</option>
        <option>Test 2</option>
        <option>Test 3</option>
      </datalist>
    </>
  );
}
```

Within my SearchInput.jsx I can't seem to be able to use `onChange={(event) => onTextChange(event)}`
This throws an error that it is undefined. I'm unsure on how to get this working.